data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/SourceInfoProvider.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.opensearch;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Map;
+
+
+public class SourceInfoProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SourceInfoProvider.class);
+
+    private String datasource;
+
+    private static final String GET_REQUEST_MEHTOD = "GET";
+
+    private static final String CONTENT_TYPE = "content-type";
+
+    private static final String CONTENT_TYPE_VALUE = "application/json";
+
+    private static final String VERSION = "version";
+
+    private static final String DISTRIBUTION = "distribution";
+
+    private static final String ELASTIC_SEARCH = "elasticsearch";
+
+    private static final String CLUSTER_STATS_ENDPOINTS = "_cluster/stats";
+
+    private static final String CLUSTER_HEALTH_STATUS = "status";
+
+    private static final String CLUSTER_HEALTH_STATUS_RED = "red";
+
+    private static final String NODES = "nodes";
+
+    private static final String VERSIONS = "versions";
+
+    /**
+     *
+     *
+      * @param openSearchSourceConfiguration
+     * @return
+     * This method will help to identify the source information eg(opensearch,elasticsearch)
+     */
+      public String getSourceInfo(final OpenSearchSourceConfiguration openSearchSourceConfiguration) {
```

Comment:
Do we need this method? I would think we would need one method in this provider to build a ClusterInfo object. The object should be re-usable.